drivers/mtd/mtd.c

```diff
@@ -42,12 +43,57 @@ int mtd_read(mtd_dev_t *mtd, void *dest, uint32_t addr, uint32_t count)
         return -ENODEV;
     }
 
-    if (mtd->driver->read) {
-        return mtd->driver->read(mtd, dest, addr, count);
+    /* page size is always a power of two */
+    const uint32_t page_shift = 32 - __builtin_clz(mtd->page_size) - 1;
```

Comment:
I guess this should be something more portable.

Reply:
`__builtin_clz` is provided by a software implementation if no `clz` instruction is available.
Or do you suggest a non-power 2 page size might be possible?

Reply:
> `__builtin_clz` is provided by a software implementation if no `clz` instruction is available.

That also depends on the compiler used right?

Why not just use `bitarithm_lsb()`?